### DISCUSSION TITLE
[Win7] VirtualBox installation fails if Python not installed

### DISCUSSION THREAD

**Discussion by Ryusennin** [BOINC/boinc#5296] [General]
The last Boinc package 7.22.2 has several user-friendliness issues.

Contrary to what the web page states, the installer comes with VirtualBox 7 and NOT VirtualBox 6. Vbox7 requires Python to be installed but this is mentioned nowhere, not even during the installation process, and the VirtualBox installation will silently fail without any error message. The user will then wonder why Boinc keeps complaining about "VirtualBox not installed".

The second related issue is specific to Windows 7. **The last working Python version for Win7 is 3.8.10**, but the available build is buried in a huge list of downloads which the average user will never sift through. The Boinc installer should be better at mentioning (or even taking care of) the specific requirements.

https://www.python.org/downloads/windows/

**Comment by AenBleidd** [comment]
@Ryusennin,

> Contrary to what the web page states, the installer comes with VirtualBox 7 and NOT VirtualBox 6.

That is an error on the description page. Thank you for mentioning this, will be fixed.

> Vbox7 requires Python to be installed but this is mentioned nowhere, not even during the installation process, and the VirtualBox installation will silently fail without any error message.

As you correctly pointed, this is the requirement of the VBox. This is a good point, BOINC installer should show an installation error from the VBox installer if any happened.
I doubt current installer will be changed to fix that issue, but we will manage this case in the new installer when ready.

> The second related issue is specific to Windows 7.

Windows 7, Windows 8 and Windows 8.1 are not supported by the Microsoft anymore. If you still use it - I strongly recommend to upgrade to at least Windows 10, because otherwise your OS will be vulnerable, and it's not safe to use them anymore.
We (BOINC team) still support all the Windows OSs starting from Windows 7, but as I mentioned above, since these OSs are not supported by Microsoft anymore, we will not make any fixes specific to these OSs as well.
So we will continue this support until we face some case when we will have to drop their support for some particular reason (e.g. upgrading to the more safe Win32 API as it was done earlier, when we dropped support of Windows XP).

> The Boinc installer should be better at mentioning (or even taking care of) the specific requirements.

As it was mentioned above, BOINC doesn't require Python to be installed. This is a VBox requirement, and we will try to handle this in a proper way.
